### PR TITLE
fix struct name output by only returning last name segment

### LINKED ISSUE #4624: `cast interface` uses wrong name / scope for structs
### Component

Cast

### Have you ensured that all of these are up to date?

- [x] Foundry
- [x] Foundryup

### What version of Foundry are you on?

cast 0.2.0 (da2392e 2023-03-22T00:04:12.437988019Z)

### What command(s) is the bug in?

cast interface

### Operating System

Linux

### Describe the bug

When generating an interface for a contract that uses structs, the structs are not renamed / scoped to the name of the generated interface.

This occurs both when passing a name manually via the `-n` option or when using the default `Interface` name.

Example:

```solidity
pragma solidity ^0.8.10;

interface Interface {
    struct Foo {
        uint256 a;
        uint256 b;
    }

	# EXPECTED:
    function bar() external returns (Interface.Foo memory);
    
    # ACTUAL:
    function bar() external returns (OriginalName.Foo memory);
}
```

### PR DESCRIPTION
**Pull request by fubhy**
## Motivation

Fixes #4624 

## Solution

I'm not sure if this is the right approach but it *should* be sufficient to just output the struct name without the parent scope as we are generated single, flat interfaces here. Or am I missing sth.?

### CODE REVIEW

File: utils/src/abi.rs [hunk 1]
```diff
@@ -152,7 +152,7 @@ fn expand_input_param_type(
             let ty = if let Some(struct_name) =
                 structs.get_function_input_struct_solidity_id(&fun.name, param)
             {
-                struct_name.to_string()
+                struct_name.split('.').last().unwrap().to_string()
```
Comment (mattsse) [inline utils/src/abi.rs:155]:
```suggestion
                struct_name.rsplit('.').next().unwrap().to_string()
```

File: utils/src/abi.rs [hunk 2]
```diff
@@ -187,7 +187,7 @@ fn expand_output_param_type(
                     &fun.name,
                     param.internal_type.as_ref().unwrap(),
                 ) {
-                    struct_name.to_string()
+                    struct_name.split('.').last().unwrap().to_string()
```
Comment (mattsse) [inline utils/src/abi.rs:190]:
```suggestion
                    struct_name.rsplit('.').next().unwrap().to_string()
```

**Review by mattsse — CHANGES_REQUESTED** [review]
ty, this is the right fix because this function returns the internal identifier and not the struct name, this is safe to unwrap because non empty string

smol nit

**Comment by fubhy** [comment]
@mattsse Thanks for the lightning fast review lol

**Review by mattsse — APPROVED** [review]

**Pull request merged**